irohad/ordering/impl/on_demand_os_client_grpc.cpp

```diff
@@ -22,26 +23,28 @@ OnDemandOsClientGrpc::OnDemandOsClientGrpc(
       time_provider_(std::move(time_provider)),
       proposal_request_timeout_(proposal_request_timeout) {}
 
-void OnDemandOsClientGrpc::onTransactions(transport::Round round,
-                                          CollectionType transactions) {
-  proto::TransactionsRequest request;
+void OnDemandOsClientGrpc::onBatches(consensus::Round round,
+                                     CollectionType batches) {
+  proto::BatchesRequest request;
   request.mutable_round()->set_block_round(round.block_round);
   request.mutable_round()->set_reject_round(round.reject_round);
-  for (auto &transaction : transactions) {
-    *request.add_transactions() = std::move(
-        static_cast<shared_model::proto::Transaction *>(transaction.get())
-            ->getTransport());
+  for (auto &batch : batches) {
+    for (auto &transaction : batch->transactions()) {
+      *request.add_transactions() = std::move(
+          static_cast<shared_model::proto::Transaction *>(transaction.get())
```

Comment:
Maybe `static_pointer_cast`?

Reply:
It will construct an additional `shared_ptr`, while we are trying to move `shared_ptr`s where possible. I do not think there will be a benefit here.